webapp/src/app/shared/widgets/npc-states-widget/npc-states/npc-states.component.ts

```diff
@@ -42,19 +47,27 @@ export class NpcStatesComponent implements OnInit {
 		this.selectTab(this.index);
 	}
 	
+	ngDoCheck() {
```

Comment:
Don't use this function without a very good reason. It's only there to manually detect changes and gets called way too often.
Recalculating warnings can be triggered manually instead